src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
+    }
+    public static void parseInput(int countTasks, Task[] tasks) {
+
+        //Input and read inputs from the user
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+        String[] words = line.split(" ");
+        String command = words[0];
+
+        while (true) {
+            //System.out.println(command);
+            if (command.equalsIgnoreCase("bye")) {
+                return;
+            } else if (command.equalsIgnoreCase("list")) {
+                System.out.println(HORIZONTAL);
+                printTasks(countTasks, tasks);
+                System.out.println(HORIZONTAL);
+            } else if (command.equalsIgnoreCase("todo")) {
+                String newTodo = line.substring(TODO_INDEX);
+                countTasks = addTodos(countTasks, tasks, newTodo);
+                System.out.println("\tGot it. I've added this task: ");
+                System.out.println("\t"+tasks[countTasks - 1].getType() + tasks[countTasks - 1].getStatusIcon() + " " + tasks[countTasks - 1].getDescription());
+                System.out.println("\tNow you have " + countTasks + " tasks in the list.");
+            } else if (command.equalsIgnoreCase("deadline")) {
+                if (line.indexOf("/by") == -1) {
+                    System.out.println("Invalid command for deadline. Must be \n deadline <nameOfEvent> /at <time>");
+                }
+                else {
+                    String newDL = line.substring(EVENT_INDEX, line.indexOf("/by"));
+                    String time = line.substring(line.indexOf("/by") + 3);
+                    countTasks = addDeadlines(countTasks, tasks, newDL, time);
+                    System.out.println("\tGot it. I've added this task: ");
+                    System.out.println("\t" + tasks[countTasks - 1].getType() + tasks[countTasks - 1].getStatusIcon() + " " + tasks[countTasks - 1].getDescription() + " by:" + tasks[countTasks - 1].getTime());
+                    System.out.println("\tNow you have " + countTasks + " tasks in the list.");
+                }
+            } else if (command.equalsIgnoreCase("event")) {
+                if (line.indexOf("/at") == -1) {
+                    System.out.println("Invalid command for event. Must be \n event <nameOfEvent> /at <time>");
+                }
+                else {
+                    String newEV = line.substring(EVENT_INDEX, line.indexOf("/at"));
+                    String time = line.substring(line.indexOf("/at") + 3);
+                    countTasks = addEvents(countTasks, tasks, newEV, time);
+                    System.out.println("\tGot it. I've added this task: ");
+                    System.out.println("\t" + tasks[countTasks - 1].getType() + tasks[countTasks - 1].getStatusIcon() + " " + tasks[countTasks - 1].getDescription() + " at:" + tasks[countTasks - 1].getTime());
+                    System.out.println("\tNow you have " + countTasks + " tasks in the list.");
```

Comment:
Perhaps this could go under a function called addTask(). Will make the program more readable and modular.

Reply:
ok, brought it to addTodos, addDeadlines and addEvents